gitmod/matomo.js

```diff
@@ -1,33 +1,72 @@
 addGainsight();
 var checkRequiredElementsExist = setInterval(function () {
-  if (window.gl !== 'undefined' && document.readyState == "complete" && document.querySelectorAll('[data-project]').length) {
-    clearInterval(checkRequiredElementsExist);
-    hideThings();
-    gainsightIdentify();
+    if (window.gl !== 'undefined' && document.readyState == "complete" && document.querySelectorAll('[data-project]').length) {
+      clearInterval(checkRequiredElementsExist);
+      hideThings();
+      gainsightIdentify();
+    }
+  }, 100);
+
+var oldURL = window.location.href;
+setInterval(checkURLchange, 1000);
+
+/**
+ * Check if there is change in URL. If the change observed invoke hideThings();
+ *
+ */ 
+function checkURLchange(){
+  if(window.location.href != oldURL){
+      oldURL = window.location.href;
+      hideThings();
   }
-}, 100);
+}
 
+/**
+ * Add logic to hide the webide and edit options from Code Studio UI
+ *
+ */ 
 function hideThings () {
-  var webIdeButton = document.querySelector('[data-qa-selector="action_dropdown"]')
-  if(webIdeButton){
-    webIdeButton.setAttribute('style', 'display:none !important')
-  }
-  var editWebButton = document.querySelector('[data-qa-selector="webide_menu_item"]')
-  if (editWebButton) {
-    editWebButton.setAttribute('style', 'display:none !important')
+  // Fetch the document that contains 'Web IDE' text
+  var webIde = document.evaluate("//span[contains(., 'Web IDE')]", document, null, XPathResult.ANY_TYPE, null );
+  var webIdeDoc = webIde.iterateNext();
+  // Assign the document to webIdeDiv
+  var webIdeDiv = webIdeDoc;
+  var content;
+  var hasText;
+  // Iterate to get the parent node of the webIdeDiv, length is set to 10 as the currently
+  // it takes 9 iterations to get the parent div where we want to apply style
```

Comment:
This feels too fragile. If there is a Gitlab update to the DOM at all, this will break. Can't we use ` .closest` or something to find the correct parent div with a certain class or attribute?

Reply:
Closest will not give the parent div which we want to apply the style
applying style on li element is not giving better UX